src/Event.cpp

```diff
@@ -25,52 +25,82 @@ void Event::add_value(std::string name, std::string&& arg) {
     string_values_.emplace(name, arg);
 }
 
+//get_value
+//Params: name of event, address to store result at, bool for error checking. if true and an error occurs then exit program
 template<>
-int Event::get_value(const std::string& name, int otherwise) const {
+bool Event::get_value(const std::string& name, std::string* value, bool crash_on_fail) const {
+    const auto val = string_values.find(name); //val - pointer to the event "name"
+
+    if (val == string_values.end()) {
+        std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
+    
+        if ( crash_on_fail == true) {   //if there is an error and crash is set to true crash program
```

Comment:
Inconsistent spacing between the left bracket '(' and 'crash_on_fail'. This is repeated several times. Please fix for all of them.